controllers/ajax.py

```diff
@@ -219,24 +219,23 @@ def gethist():
     """
     codetbl = db.code
     acid = request.vars.acid
-    sid = request.vars.sid
 
-    if sid:
-        query = ((codetbl.sid == sid) & (codetbl.acid == acid) & (codetbl.timestamp != None))
+    if request.vars.sid:
+        sid = request.vars.sid
+    elif auth.user.username:
+        sid = auth.user.username
     else:
-        if auth.user:
```

Comment:
I don't think this should be removed.   If the sid is a parameter then we are getting the history for that sid from the grading interface.  But if a student is logged in then we get the sid from their logged in username.  How will this work if this is deleted?